WalletWasabi/Services/MemPoolService.cs

```diff
@@ -222,12 +228,18 @@ public async Task<bool> TryPerformMempoolCleanupAsync(NodesGroup nodes, Cancella
 					}
 				}
 
-				uint256[] toRemove = TransactionHashes.Except(allTxs).ToArray();
-				foreach (uint256 tx in toRemove)
+				int removedTxCount = 0;
+				foreach (uint256 tx in TransactionHashes.ToArray())
 				{
-					TransactionHashes.TryRemove(tx);
+					if (!allTxs.Contains(tx))
```

Comment:
I think this should be less optimal. The `Except` is a set operation, which should be faster than going through everything and look at contains, no?

Reply:
I don't know, I should measure it. Anyway, I don't know how it could be faster.

Reply:
I left it to be your contains.